### PR TITLE
[CI] Add markdown check for normalized code-blocks + .md page fixes

### PR DESCRIPTION
**Pull request by chalin**
- Contributes to #3122
- This PR is best reviewed by **ignoring whitespace changes**
- Adds an NPM script named `fix:markdwon` which will run markdownlint in "fix" mode
- Adds a custom markdownlint rule, named `trim-code-block-and-unindent`, to check and optionally clean up fenced code blocks of:
  - Excessive blank lines at the start or end of the code
  - Unnecessary indentation
- Adds `trim-code-block-and-unindent:true` and `no-trailing-spaces:true` to the markdownlint config
- Fixes fenced code blocks in our docs using the new rule
- **Notes**:
  - Custom rules are controlled in the same way as other rules. So they can be turned on/off for the project or selectively within a file.
  - The added custom rule is a bit hacky (I didn't quite understand how markdownlint rules worked when I started 🤷🏼‍♂️), but it works so I'm committing it now. I'll rework the rule later.

### CODE REVIEW

**Review by cartermp — COMMENTED** [review]
Hmmm. What is "unecessary indent" here for code blocks? Maybe a sample would help demonstrate. I don't understand offhand how this accounts for whitespace-sensitive languages.

**Comment by chalin** [comment]
Maybe "excessive indentation" would be a better term than "unnecessary indentation"?

Here's an example from the PR:

Screenshot 2023-08-15 at 08 51 13

The algorithm will shift the code in the code block to the left as much as possible while preserving the relative indentation of non-whitespace characters. Of the 25+ files in this PR that were fixed, do you see any that were incorrectly adjusted (I've re-enabled view-whitespace diffs).

**Review by ferd — APPROVED** [review]

**Review by cartermp — APPROVED** [review]
Okay, thanks, makes sense. So long as relative indentation is preserved then this is good.

File: content/en/docs/demo/services/shipping.md [hunk 1]
```diff
@@ -142,22 +142,22 @@ See below for a snippet from `shiporder` that holds context and a span in scope.
...
-    let tracer = global::tracer("shippingservice");
-    let mut span = tracer
-        .span_builder("hipstershop.ShippingService/ShipOrder").with_kind(SpanKind::Server).start_with_context(&tracer, &parent_cx);
+let parent_cx =
+global::get_text_map_propagator(|prop| prop.extract(&MetadataMap(request.metadata())));
+// in this case, generating a tracking ID is trivial
+// we'll create a span and associated events all in this function.
+let tracer = global::tracer("shippingservice");
+let mut span = tracer
+    .span_builder("hipstershop.ShippingService/ShipOrder").with_kind(SpanKind::Server).start_with_context(&tracer, &parent_cx);
```
[the remark this answers is not in the document]
Comment (chalin) [reply content/en/docs/demo/services/shipping.md:151]: @julianocosta89 - FYI, I'll add your code change to a followup PR.

[the remark this answers is not in the document]
Comment (chalin) [reply content/en/docs/demo/services/shipping.md:151]: Let's track this code change via:

- #3179

**Pull request merged**